annotated_types/__init__.py

```diff
@@ -205,6 +205,28 @@ def __iter__(self) -> Iterator[BaseMetadata]:
         if self.le is not None:
             yield Le(self.le)
 
+    def __check(self, other, which):
```

Comment:
```suggestion
    def __check(self, other: Literal["lo", "hi"], which: Literal["lo", "hi"]) -> None:
```

Is that right? Wither way, a type hint here would be useful for readers.